stablehlo/dialect/TypeInference.cpp

```diff
@@ -4370,17 +4278,16 @@ LogicalResult verifySortOp(std::optional<Location> location, ValueRange inputs,
   auto operandTypes = inputs.getTypes();
   for (auto operandType : operandTypes) {
     auto operandShapedType = operandType.cast<ShapedType>();
-    if (operandShapedType.hasRank()) {
-      int64_t cmpDim = dimension;
-      int64_t rank = operandShapedType.getRank();
-      // sort_c4
-      if (cmpDim < -rank || cmpDim >= rank)
-        return emitOptionalError(
-            location, "dimension attribute value must be in range [-", rank,
-            ", ", rank, "), but found ", cmpDim);
-      else
-        break;  // ODS SameOperandsAndResultShape asserts inputs have same shape
-    }
+    int64_t cmpDim = dimension;
+    int64_t rank = operandShapedType.getRank();
+    // sort_c4
+    if (cmpDim < -rank || cmpDim >= rank)
+      return emitOptionalError(location,
+                               "dimension attribute value must be in range [-",
+                               rank, ", ", rank, "), but found ", cmpDim);
+    else
+      break;  // ODS SameOperandsAndResultShape asserts inputs have same
```

Comment:
Nit: Not directly related to your changes, but let's remove the `else` keyword here and move the comment above the `break`

Reply:
Done.